### PR TITLE
[#IR-309] As a dev, Added update password page and integrate it

### PR DESCRIPTION
**Pull request by chahat-arora**

### CODE REVIEW

**Comment by himanshu-garg-165** [comment]
Please do the following:
- Add cancel button on every step for password reset, which would redirect the user to the home page.

File: src/app/core/route-guards/auth.guard.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  constructor(
+    private router: Router,
+    private authService: AuthService,
+  ) { }
+
+  canActivate(
+    next: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot): Observable<boolean> | Promise<boolean> | boolean {
+
+    if (!this.authService.getEmailAndReSendTime().email || !this.authService.getEmailAndReSendTime().reSendTime) {
```
Comment (himanshu-garg-165) [inline src/app/core/route-guards/auth.guard.ts]: We should evaluate `this.authService.getEmailAndReSendTime()` one time and use it.

Comment (chahat-arora) [reply src/app/core/route-guards/auth.guard.ts]: done.

File: src/app/core/route-guards/auth.guard.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+export class AuthGuard implements CanActivate {
+
+  constructor(
+    private router: Router,
+    private authService: AuthService,
+  ) { }
+
+  canActivate(
+    next: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot): Observable<boolean> | Promise<boolean> | boolean {
```
[the remark this answers is not in the document]
Comment (chahat-arora) [reply src/app/core/route-guards/auth.guard.ts]: done.

File: src/app/core/route-guards/auth.guard.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import { Injectable } from '@angular/core';
+import { CanActivate, ActivatedRouteSnapshot, RouterStateSnapshot, Router } from '@angular/router';
+import { Observable } from 'rxjs/Observable';
```
[the remark this answers is not in the document]
Comment (chahat-arora) [reply src/app/core/route-guards/auth.guard.ts]: done.

File: src/app/account/update-password/update-password.component.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+@Component({
+  selector: 'app-update-password',
+  templateUrl: './update-password.component.html',
+  styleUrls: ['./update-password.component.scss']
+})
+export class UpdatePasswordComponent {
+
+  public passwordGroup = new FormGroup({
+    firstPassword: new FormControl('', [Validators.required, Validators.minLength(8)]),
+    secondPassword: new FormControl('', [Validators.required, Validators.minLength(8)])
```
[the remark this answers is not in the document]
Comment (chahat-arora) [reply src/app/account/update-password/update-password.component.ts]: done.

File: src/app/account/update-password/update-password.component.ts [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+@Component({
+  selector: 'app-update-password',
+  templateUrl: './update-password.component.html',
+  styleUrls: ['./update-password.component.scss']
+})
+export class UpdatePasswordComponent {
+
+  public passwordGroup = new FormGroup({
+    firstPassword: new FormControl('', [Validators.required, Validators.minLength(8)]),
```
[the remark this answers is not in the document]
Comment (chahat-arora) [reply src/app/account/update-password/update-password.component.ts]: done.

File: src/app/account/update-password/update-password.component.ts [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    secondPassword: new FormControl('', [Validators.required, Validators.minLength(8)])
+  }, {
+      validators: this.isPasswordSame
+    });
+  public hideFirstPassword = true;
+  public hideSecondPassword = true;
```
[the remark this answers is not in the document]
Comment (chahat-arora) [reply src/app/account/update-password/update-password.component.ts]: done.

File: src/app/account/update-password/update-password.component.ts [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+})
+export class UpdatePasswordComponent {
+
+  public passwordGroup = new FormGroup({
+    firstPassword: new FormControl('', [Validators.required, Validators.minLength(8)]),
+    secondPassword: new FormControl('', [Validators.required, Validators.minLength(8)])
+  }, {
+      validators: this.isPasswordSame
+    });
+  public hideFirstPassword = true;
```
[the remark this answers is not in the document]
Comment (chahat-arora) [reply src/app/account/update-password/update-password.component.ts]: done.

File: src/app/account/update-password/update-password.component.ts [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+      .subscribe(() => {
+        this.router.navigate([APP_ROUTE_URLS.login]);
+      },
+        (errorResponse: any) => {
+          if (errorResponse.status === 400) {
+            this.errorMessage = errorResponse.data.error;
+          } else {
+            this.errorMessage = LOGIN_ERROR_MESSAGES.internalError;
+          }
+          this.showError = true;
```
Comment (himanshu-garg-165) [inline src/app/account/update-password/update-password.component.ts]: Why do we need this? Is `errorMessage` not sufficient for the same?

File: src/app/core/route-guards/auth.guard.ts [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+    state: RouterStateSnapshot,
+    ): Observable<boolean> | Promise<boolean> | boolean {
+
+    const emailAndTimeData = this.authService.getEmailAndReSendTime();
+
+    if (!emailAndTimeData.email || !emailAndTimeData.reSendTime) {
+      this.router.navigateByUrl(APP_ROUTE_URLS.forwardSlash);
+      return false;
+    }
+    if (next.routeConfig.path === APP_ROUTE_URLS.updatePassword && !this.authService.getOTP()) {
```
Comment (himanshu-garg-165) [inline src/app/core/route-guards/auth.guard.ts]: Why do we need to check the next route path?

File: src/app/account/update-password/update-password.component.ts (hunk 8 above)
Comment (chahat-arora) [reply src/app/account/update-password/update-password.component.ts]: done.

File: src/app/core/route-guards/auth.guard.ts (hunk 9 above)
Comment (chahat-arora) [reply src/app/core/route-guards/auth.guard.ts]: discussed.

**Pull request merged**